pkg/proc/proc.go

```diff
@@ -226,6 +226,9 @@ func Continue(dbp Process) error {
 				return conditionErrors(threads)
 			case g == nil || dbp.Common().fncallForG[g.ID] == nil:
 				// a hardcoded breakpoint somewhere else in the code (probably cgo)
+				if !curthread.Arch().BreakInstrMovesPC() {
+					curthread.SetPC(loc.PC + uint64(curthread.Arch().BreakpointSize()))
```

Comment:
IIRC this case also happens when the target process is stopped manually, in that case the instruction at PC won't be a hardcoded breakpoint but some random instruction so we can't blindly advance by BreakpointSize. This code should read the memory at loc.PC with dbp.CurrentThread().ReadMemory and compare it to dbp.BinInfo().Arch.BreakpointInstruction()

Reply:
AYS? The comment here said "a hardcoded breakpoint somewhere else".

Reply:
modified.

Reply:
> AYS? The comment here said "a hardcoded breakpoint somewhere else".

Yes, the comment is wrong (or, rather, incomplete).